### DISCUSSION TITLE
How can I set the pollutant concentration of the node?

### DISCUSSION THREAD

**Discussion by Escaper29** [pyswmm/pyswmm#387] [Q&A] [answered]
I want to use the function of setting node pollutant concentration to Implementation the treatment function in swmm, but I find that the pollutant setter in node does not work sometimes. It looks like it is related to the attributes of the node. I want to ask what needs to be noticed about the pollutant setter in node.

**Comment by karosc** [comment, marked as answer]
We should probably add this to the documentation, but for the pollutant setter to work, the node and pollutant must be present in the treatment section of the INP file even if the function is meaningless having no rate constant. [See my comment in this swmm-python pr](https://github.com/OpenWaterAnalytics/swmm-python/pull/94#issuecomment-1013951491) for more info

Try doing that and see if it helps you.
